src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -37,7 +40,35 @@ public record CollectionSettings(
     String modelName,
     IndexingConfig indexingConfig) {
 
-  public record IndexingConfig(Set<String> allowed, Set<String> denied) {
+  private static final CollectionSettings EMPTY =
+      new CollectionSettings("", false, 0, null, null, null, null);
+
+  public static CollectionSettings empty() {
+    return EMPTY;
+  }
+
+  public DocumentProjector indexingProjector() {
+    // IndexingConfig null if no indexing definitions: default, index all:
+    if (indexingConfig == null) {
+      return DocumentProjector.identityProjector();
+    }
+    // otherwise get lazily initialized indexing projector from config
+    return indexingConfig.indexingProjector();
+  }
+
+  public record IndexingConfig(
+      Set<String> allowed, Set<String> denied, Supplier<DocumentProjector> indexedProject) {
+    public IndexingConfig(Set<String> allowed, Set<String> denied) {
+      this(
+          allowed,
+          denied,
+          Suppliers.memoize(() -> DocumentProjector.createForIndexing(allowed, denied)));
+    }
+
+    public DocumentProjector indexingProjector() {
+      return indexedProject.get();
```

Comment:
This does build the Projector for every call. Is the idea to get this once use it in code passing everywhere?

Reply:
No, it uses Guava's memoize() to avoid just that.

Reply:
I'll try to come up with a test to verify that we only get one instance; maybe `CreateCollectionCommandResolverTest`?

Reply:
Ok, easier to verify with stand-alone unit test: see `CollectionSettingsTest` which verifies that instance is created dynamically, and then same instance returned for further calls to get instance.